ccd/models/lasso.py

```diff
@@ -29,17 +29,26 @@ def coefficient_matrix(dates, avg_days_yr, num_coefficients):
 
     matrix = np.zeros(shape=(len(dates), 7), order='F')
 
+    # lookup optimizations
+    # Before optimization - 12.53% of total runtime
+    # After optimization  - 10.57% of total runtime
+    w12 = w * dates
```

Comment:
If there are different names you'd prefer or what I submitted isn't clear let me know.